### ISSUE TITLE
Compilation of rune 0.7.0 fails due to codespan_reporting

### ISSUE THREAD

**Issue by PatchMixolydic** [rune-rs/rune#238]
I tried the following code:

`src/main.rs`
```rust
fn main() {}
```

`Cargo.toml`
```toml
[package]
name = "rune_issue"
version = "0.1.0"
authors = ["Ruby Lazuli"]
edition = "2018"

[dependencies]
rune = "0.7.0"
```

I expected the code to compile with no errors and do nothing.

Instead, this happened:
```
$ cargo run
...
   Compiling runestick v0.7.0
   Compiling rune v0.7.0
error[E0432]: unresolved import `codespan_reporting::diagnostic`
  --> /home/sparkpin/.cargo/registry/src/github.com-1ecc6299db9ec823/rune-0.7.0/src/diagnostics.rs:15:25
   |
15 | use codespan_reporting::diagnostic::{Diagnostic, Label};
   |                         ^^^^^^^^^^ could not find `diagnostic` in `codespan_reporting`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0432`.
error: could not compile `rune`
```

Tested rustc versions:
```
rustc 1.48.0 (7eac88abb 2020-11-16)
binary: rustc
commit-hash: 7eac88abb2e57e752f3302f02be5f3ce3d7adfb4
commit-date: 2020-11-16
host: x86_64-unknown-linux-gnu
release: 1.48.0
LLVM version: 11.0
```
```
rustc 1.51.0-nightly (257becbfe 2020-12-27)
binary: rustc
commit-hash: 257becbfe4987d1f7b12af5a8dd5ed96697cd2e8
commit-date: 2020-12-27
host: x86_64-unknown-linux-gnu
release: 1.51.0-nightly
```

**Comment by udoprog** [comment]
Hey!

This is a bit unfortunate. This should be fixed by the next release. But I'll leave this open if anyone is interested in backporting a fix to `0.7.1`.

For now, it's generally recommended to use it of git. A lot of things are still changing between releases:
```toml
rune = { git = "https://github.com/rune-rs/rune" }
```

**Comment by udoprog** [comment]
Closing this now since 0.7 is about a year old.
